packages/router-core/src/load-matches.ts

```diff
@@ -369,7 +369,11 @@ const executeBeforeLoad = (
   const parentMatchContext =
     parentMatch?.context ?? inner.router.options.context ?? undefined
 
-  const context = { ...parentMatchContext, ...match.__routeContext }
+  const context = {
+    ...match.__beforeLoadContext,
```

Comment:
this is not correct IMO as this will feed the previous beforeLoad context into beforeLoad again.